resolver/resolver.go

```diff
@@ -311,15 +287,9 @@ func (res *Resolver) HandleNonMesos(w dns.ResponseWriter, r *dns.Msg) {
 		// set refused
 		m.SetRcode(r, 5)
 	} else {
-
-		proto := "udp"
-		if _, ok := w.RemoteAddr().(*net.TCPAddr); ok {
```

Comment:
Taking a protocol into account means that we have to create a new client on every request. If we think about it, clients can't possibly depend on the outgoing recursion transport protocol. That's entirely transparent to them. What I think we should do, instead, is to attempt to exchange the same message with both UDP and, if that fails, due to issues such as excessive payload size, TCP.